lib/utils.js

```diff
@@ -208,4 +208,105 @@ exports.createNullProtoObjWherePossible = (function () {
   };
 })();
 
+// Embedded argument parser from https://github.com/jakejs/jake/blob/master/lib/parseargs.js
+// See original source for doc.
+// Fixes #510 and allows removing `jake` as a prod dependency.
+exports.parseArgs = function parseArgs(programOpts, programArgs) {
+  let isOpt = function (arg) {
+    return arg.indexOf('-') === 0;
+  };
+  let removeOptPrefix = function (opt) {
+    return opt.replace(/^--/, '').replace(/^-/, '');
+  };
+
+  const Parser = function (opts) {
+    this.opts = {};
+    this.taskNames = null;
+    this.envVars = null;
+    this.reg = opts;
+    this.shortOpts = {};
+    this.longOpts = {};
+
+    let self = this;
+    [].forEach.call(opts, function (item) {
+      self.shortOpts[item.abbr] = item;
+      self.longOpts[item.full] = item;
+    });
+  };
+
+  Parser.prototype = new (function () {
+    let _trueOrNextVal = function (argParts, args) {
+      if (argParts[1]) {
+        return argParts[1];
+      } else {
+        return !args[0] || isOpt(args[0]) ? true : args.shift();
+      }
+    };
+
+    this.parse = function (args) {
+      let cmds = [];
+      let cmd;
+      let envVars = {};
+      let opts = {};
+      let arg;
+      let argItem;
+      let argParts;
+      let cmdItems;
+      let taskNames = [];
+      let preempt;
 
+      while (args.length) {
+        arg = args.shift();
+
+        if (isOpt(arg)) {
+          arg = removeOptPrefix(arg);
+          argParts = arg.split('=');
+          argItem = this.longOpts[argParts[0]] || this.shortOpts[argParts[0]];
+          if (argItem) {
```

Comment:
If this condition does not meet, I. E. this is an unrecognized option, do we want to throw an error?

See https://github.com/mde/ejs/issues/690